### PR TITLE
Allow to detect/obtain breaker from hook dispatcher

### LINKED ISSUE #138: Hook break can not be detected
See https://github.com/atk4/core/blob/2d72fedda6604b0f28f9c128acc506f0a5993f23/src/HookTrait.php#L156

Currently, if breaker returns an array the break by hook breaker can not be detected.

Why not to throw an exception? If circuit breaker is desired, it should alway return an array and reference parameter like `bool &$wasInterrupted` should be added to the hook execution method. Or return an CircuitBreaker object?

### PR DESCRIPTION
**Pull request by mvorisek**
fix #138

### CODE REVIEW

File: src/HookBreaker.php [hunk 1]
```diff
@@ -12,14 +12,22 @@ class HookBreaker extends Exception
     /**
      * @var mixed
      */
-    public $return_value;
+    protected $returnValue;
```
Comment (mvorisek) [inline src/HookBreaker.php:15]: Full BC as `HookBreaker` was previously never returned.

File: src/HookTrait.php [hunk 2]
```diff
@@ -173,24 +175,20 @@ public function hook(string $spot, array $args = [])
...
-                        $return[$index] = call_user_func_array(
-                            $data[0],
-                            array_merge(
-                                [$this],
-                                $args,
-                                $data[1]
-                            )
-                        );
+                    foreach ($_data as $index => $data) {
+                        $return[$index] = $data[0](...array_merge(
```
Comment (DarkSide666) [inline src/HookTrait.php:179]: Isn't it possible that $data[0] is defined in array format like `[$this, 'methodName']` ?

File: src/SessionTrait.php [hunk 3]
```diff
@@ -69,7 +69,7 @@ public function memorize(string $key, $value)
         $this->startSession();
 
         if (is_callable($value)) {
-            $value = call_user_func($value, $key);
+            $value = $value($key);
```
Comment (DarkSide666) [inline src/SessionTrait.php:72]: Isn't it possible that $value is defined in array format like `[$this, 'methodName']` ?

File: src/SessionTrait.php [hunk 4]
```diff
@@ -109,7 +109,7 @@ public function recall(string $key, $default = null)
             || $_SESSION[$this->session_key][$this->name][$key] === null
         ) {
             if (is_callable($default)) {
-                $default = call_user_func($default, $key);
+                $default = $default($key);
```
Comment (DarkSide666) [inline src/SessionTrait.php:112]: same here

File: src/HookTrait.php (hunk 2 above)
Comment (mvorisek) [reply src/HookTrait.php:179]: https://3v4l.org/E5cZ5

Comment (DarkSide666) [reply src/HookTrait.php:179]: Oh perfect then!
I see my old-school php 5.x habits here :D

**Pull request merged**